examples/preprocessing/plot_target_encoder.py

```diff
@@ -0,0 +1,120 @@
+"""
+============================================
+Comparing Target Encoder with Other Encoders
+============================================
+
+.. currentmodule:: sklearn.preprocessing
+
+The :class:`TargetEncoder` uses the value of the target to encode each
+categorical feature. In this example, we will compare three different approaches
+for handling categorical features: :class:`TargetEncoder`,
+:class:`OrdinalEncoder`, :class:`OneHotEncoder` and dropping the category.
+For more information about the :class:`TargetEncoder` see the
+:ref:`User Guide <target_encoder>`.
+
+.. note::
+  :class:`TargetEncoder` uses a cross validation scheme in
+  :meth:`~TargetEncoder.fit_transform` to prevent leaking the target during training.
+  In :meth:`~TargetEncoder.fit_transform`, the data is split according to the `cv`
+  parameter. Categorical encodings are learned from split and used to encode the other
+  split. Afterwards, a final categorical encoding is learned from all the data, which
+  is then used to encode data during :meth:`~TargetEncoder.transform`. This means that
+  `fit(X, y).transform(X)` does not equal `fit_transform(X, y)`.
+"""
+
+# %%
+# Loading Data from OpenML
+# ========================
+# First, we load the wine reviews dataset, where the target is the points given
+# be a reviewer:
+from sklearn.datasets import fetch_openml
+
+wine_reviews = fetch_openml(data_id=42074, as_frame=True, parser="liac-arff")
+
+df = wine_reviews.frame
+df.head()
+
+# %%
+# For this example, we use the following subset of numerical and categorical
+# features in the data. The target are continuous values from 80 to 100:
+numerical_features = ["price"]
+categorical_features = [
+    "country",
+    "province",
+    "region_1",
+    "region_2",
+    "variety",
+    "winery",
+]
+target_name = "points"
+
+X = df[numerical_features + categorical_features]
+y = df[target_name]
+
+y.hist()
+
+# %%
+# Then, we split the dataset into a training and test set.
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
+
+print(f"Samples in training set: {len(X_train)}\nSamples in test set: {len(X_test)}")
+
+# %%
+# Training and Evaluating Pipelines with Different Encoders
+# =========================================================
+# In this section, we will evaluate pipelines with
+# :class:`~sklearn.ensemble.HistGradientBoostingRegressor` with different encoding
+# strategies. First, we list out the encoders we will be using to preprocess
+# the categorical features:
+from sklearn.compose import ColumnTransformer
+from sklearn.preprocessing import OrdinalEncoder
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import TargetEncoder
+
+categorical_preprocessors = [
+    ("drop", "drop"),
+    ("ordinal", OrdinalEncoder(handle_unknown="use_encoded_value", unknown_value=-1)),
+    (
+        "one_hot_encoder",
```

Comment:
For the sake of consistency with the other preprocessor names:

```suggestion
        "one_hot",
```

This might also fix the cropped label problem as a side effect (without having to use the constrained layout).